### PR TITLE
minimal technique to validate backups

### LINKED ISSUE #292: Provide a way to check backups after creation and as single command
With so many users using sd cards and that they get broken we should verify an backup directly after creation and offer this also as own cli command. 
This verification should at least verify all
Json files to make sure they are parsable.

### PR DESCRIPTION
**Pull request by foxriver76**
- as described in #292 (will close #292)
- at the moment it provides a very minimalistic way of validation
- maybe we can discuss here about a more complex validation, then I can add this to the PR

### CODE REVIEW

File: lib/setup/setupBackup.js [hunk 1]
```diff
@@ -594,17 +594,13 @@ class BackupRestore {
...
-                const backupJSON = require(tmpDir + '/backup/backup.json');
-                if (!backupJSON.objects || !backupJSON.objects.length) {
-                    console.log('Backup does not contain valid objects');
-                    this.processExit(26);
-                } // endIf
-                resolve();
-            } catch (e) {
-                reject(e);
-            } // endTryCatch
+        return new Promise(resolve => {
```
Comment (AlCalzone) [inline lib/setup/setupBackup.js]: I find it weird that you handle one error (invalid objects) directly inside the promise and the case of corrupt files in the main file. I would probably reject the promise with the correct error message in both cases and then print the message and call process.exit in the main file.

Comment (foxriver76) [reply lib/setup/setupBackup.js]: Idea was to handle errors directly where I know the error code and other errors where I have to exit with code 1 in the setup and do not have to check for if (err === 'xyz') but basically you are right and it seems inconsistent. I will change this.

Comment (foxriver76) [reply lib/setup/setupBackup.js]: I think, now I get what you mean. I've looked at the other method ;-)

**Comment by Apollon77** [comment]
Somehow I miss the try/catch in calidateBackup while requiring the json ... if file content is broken somehow an exception will be raised ... I do not know if this will then be reported to the user or the promises will magically handle it :-) A understandable error for the user would be great :-)

**Pull request merged**